app/src/main/java/zapsolutions/zap/transactionHistory/transactionDetails/OnChainTransactionDetailBSDFragment.java

```diff
@@ -0,0 +1,209 @@
+import com.google.protobuf.InvalidProtocolBufferException;
+import zapsolutions.zap.R;
+import zapsolutions.zap.util.BlockExplorer;
+import zapsolutions.zap.util.ClipBoardUtil;
+import zapsolutions.zap.util.MonetaryUtil;
+import zapsolutions.zap.util.Wallet;
+import zapsolutions.zap.util.ZapLog;
+
+import java.text.DateFormat;
+import java.util.Date;
+
+public class OnChainTransactionDetailBSDFragment extends BottomSheetDialogFragment {
+
+    public static final String TAG = OnChainTransactionDetailBSDFragment.class.getName();
+    public static final String ARGS_TRANSACTION = "TRANSACTION";
+
+    private Transaction mTransaction;
+    private TextView mTransactionDescription;
+    private TextView mChannelLabel;
+    private TextView mChannel;
+    private TextView mEventLabel;
+    private TextView mEvent;
+    private TextView mAmountLabel;
+    private TextView mAmount;
+    private TextView mFeeLabel;
+    private TextView mFee;
+    private TextView mDateLabel;
+    private TextView mDate;
+    private TextView mTransactionIDLabel;
+    private TextView mTransactionID;
+    private ImageView mTransactionIDCopyButton;
+    private TextView mAddressLabel;
+    private TextView mAddress;
+    private ImageView mAddressCopyButton;
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.bsd_on_chain_transaction_detail, container);
+
+        mTransactionDescription = view.findViewById(R.id.transactionDescription);
+
+        mChannelLabel = view.findViewById(R.id.channelLabel);
+        mChannel = view.findViewById(R.id.channel);
+        mEventLabel = view.findViewById(R.id.eventLabel);
+        mEvent = view.findViewById(R.id.event);
+        mAmountLabel = view.findViewById(R.id.amountLabel);
+        mAmount = view.findViewById(R.id.amount);
+        mFeeLabel = view.findViewById(R.id.feeLabel);
+        mFee = view.findViewById(R.id.fee);
+        mDateLabel = view.findViewById(R.id.dateLabel);
+        mDate = view.findViewById(R.id.date);
+        mTransactionIDLabel = view.findViewById(R.id.transactionIDLabel);
+        mTransactionID = view.findViewById(R.id.transactionID);
+        mTransactionIDCopyButton = view.findViewById(R.id.txIDCopyIcon);
+        mAddressLabel = view.findViewById(R.id.addressLabel);
+        mAddress = view.findViewById(R.id.address);
+        mAddressCopyButton = view.findViewById(R.id.addressCopyIcon);
+
+        ImageButton closeButton = view.findViewById(R.id.closeButton);
+        closeButton.setOnClickListener(view1 -> dismiss());
+
+        if (getArguments() != null) {
+            ByteString transactionString = (ByteString) getArguments().getSerializable(ARGS_TRANSACTION);
+
+            try {
+                bindOnChainTransaction(transactionString);
+
+            } catch (InvalidProtocolBufferException exception) {
```

Comment:
Could be simplified to `} catch (InvalidProtocolBufferException | NullPointerException exception) {`